src/libcollections/dlist.rs

```diff
@@ -921,6 +1015,48 @@ mod tests {
 
     #[test]
     #[allow(deprecated)]
+    fn test_append_mut() {
+        {
+            let mut m = DList::new();
+            let mut n = DList::new();
+            n.push_back(2i);
+            m.append_mut(&mut n);
+            assert_eq!(m.len(), 1);
+            assert_eq!(m.pop_back(), Some(2));
+            assert_eq!(n.len(), 0);
+            check_links(&m);
+        }
+        {
```

Comment:
I would appreciate comments at the top of these blocks specifying the "case" they're testing. e.g. "append empty list to non-empty list"